test/parallel/test-readable-from.js

```diff
@@ -0,0 +1,94 @@
+'use strict';
```

Comment:
I've had a busy few months and I'm probably out of the loop - but does this need a `common.crashOnUnhandledRejection`?

Reply:
No, crashing on unhandled rejections is the default, but we’d probably want to add `.then(common.mustCall())` to the Promise at the end of the file to make sure it resolves

Reply:
`common.crashOnUnhandledRejection` was removed a while ago. The unhandled rejection handler is always installed in the `common` module: 
https://github.com/nodejs/node/blob/b2308339127488b12d79f4775f83e2afca459945/test/common/index.js#L697-L702

Reply:
`.then(common.mustCall())` doesn't work here (throws that it wasn't called), while `.then(() => { var x = common.mustCall(); x(); })` works completely fine.

The test process shouldn't be able to close with pending promises though - that seems like something that should be done generally for all tests to me.

Reply:
> `.then(common.mustCall())` doesn't work here (throws that it wasn't called), while `.then(() => { var x = common.mustCall(); x(); })` works completely fine.

That probably means that the combined promise does not *actually* resolve? Can you check that?

Reply:
It's definitely resolving.

Reply:
@guybedford It does not do so locally for me – using `]).then(() => console.log('resolved'), () => console.log('rejected'));` prints nothing.

Reply:
Right, but if you change it to - 

```js
]).then(() => throw new Error('ERROR'));
```

the error will display as a test error.

Reply:
Ah, no it won't, sorry!

Reply:
Ok... so this seems bad! How do we fix it?

Reply:
I mean, I haven’t looked much at the actual code, but I guess it comes down to figuring out which Promise is the first in the chain to not resolve – I’m happy to help with that if you want

Reply:
Found the faulty test, thanks.

Reply:
```diff
diff --git a/test/parallel/test-readable-from.js b/test/parallel/test-readable-from.js
index 5b03215d32c0..64a9234d5fc3 100644
--- a/test/parallel/test-readable-from.js
+++ b/test/parallel/test-readable-from.js
@@ -1,6 +1,6 @@
 'use strict';
 
-require('../common');
+const common = require('../common');
 const { once } = require('events');
 const { Readable } = require('stream');
 const { strictEqual } = require('assert');
@@ -38,7 +38,7 @@ async function toReadableOnData() {
     strictEqual(chunk, expected.shift());
   });
 
-  await once(stream, 'finish');
+  await once(stream, 'end');
 
   strictEqual(iterations, 3);
 }
@@ -91,4 +91,4 @@ Promise.all([
   toReadableOnData(),
   destroysTheStreamWhenThrowing(),
   asTransformStream()
-]);
+]).then(common.mustCall());
```

:slightly_smiling_face: